lib/internal/url.js

```diff
@@ -1112,6 +1115,66 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  for (var n = 0; n < pathname.length; n++) {
+    if (pathname[n] === '%') {
+      var third = pathname.codePointAt(n + 2) | 0x20;
+      if ((pathname[n + 1] === '2' && third === 102) || // 2f 2F /
+          (pathname[n + 1] === '5' && third === 99)) {  // 5c 5C \
+        return new Error('Path must not include encoded \\ or / characters');
```

Comment:
Ok, that is fine. Updated the errors to `TypeError`. Went ahead and updated all of the new Errors introduced by this to `TypeError` to maintain consistency.